### ISSUE TITLE
Get document property 'Has field' variable not valid

### ISSUE THREAD

**Issue by Mattmannn** [FlutterFlow/flutterflow-issues#1144]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

TLDR: Set Variable throws 'Current variable is not valid' when trying to check for Has Field in a document custom data type (map).

### Expected Behavior

The Set Variable should accept my Has Field check in Get Document Property

### Steps to Reproduce

1. Create a Firestore project
3. Connect it to the new project
4. Add Firestore project name in FF
5. Generate config files
5. In Firestore, create a 'jobs' collection
6. Create a document e.g. 'job_id' in 
7. Add a 'status' map field
8. Add 'status1' array
9. In FlutterFlow, create a custom 'statusMap' data type
10. Add 'status1' field data type String (List)
11. Create new FlutterFlow page
12. Add a container
13. Add backend query to get the job_id document (query collection, doc from reference)
14. Add text
15. Set text from variable
16. Conditional value
17. If jobs Document -> status -> Has Field -> status1
18. Confirm

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT4oivLqy4Z1ru5E7a6Jd+9VnSQ7KUR7apEv0cpOEC0jF7KuBqQiZPTBT1VvOdfiaVNcAk2YkXwf097zhvHPCMIDAwubfos/yJB+cgn0e2itVrKuC5e0SnB6QO1QIXK51rWJ3Ap5CLRZc0Ym3TmQAeiRExHDFOu/Zwh9f6PHaOI=

### Context

I have a Firestore collection with the following data structure:
```
jobs (collection)
     [doc]
          other_fields (misc. types)
          status (map/custom data type in FF)
               other_statuses (array/string(List))
               new_created (array/string(List))
                    [0] specialist1_id (string)
                    [1] specialist2_id (string)
```
I want to display a conditional text based on whether 'new_created' contains any fields (i.e. any specialists seeing the job as 'new'). I added a backend query to get a job document from reference, and then I added conditional values to my text:
`IF 'jobs' Document 'status' (map) Data Structure Field 'new_created' List Contains Item - true`
this worked fine, but I realized I wanted to check for errors first, so I tried to add
`IF 'jobs' Document 'status' (map) Has Field 'new_created'`
to check whether this new_created status field exists at all first, before checking its contents. However, I got 'Current variable is not valid'.

### Visual documentation

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/8464443/c0267900-cfa9-4827-986a-d2ad435dcbe9)
![image](https://github.com/FlutterFlow/flutterflow-issues/assets/8464443/fde6e088-36f6-4bf0-93a8-d4ba9774ec04)
![image](https://github.com/FlutterFlow/flutterflow-issues/assets/8464443/c5c7bee6-6bdd-4f7e-9d3b-a14fb134d30a)
This is the first condition, which works fine, but then when trying to accomplish the following, it throws an error:
![image](https://github.com/FlutterFlow/flutterflow-issues/assets/8464443/df820dca-3714-4883-b8d2-7ff63ba2a864)
This is my 'jobs' collection schema:
![image](https://github.com/FlutterFlow/flutterflow-issues/assets/8464443/5b5015ae-b87b-4439-9564-f7b82073895f)
This is my 'mapStatus' custom data type:
![image](https://github.com/FlutterFlow/flutterflow-issues/assets/8464443/62f5579f-3d53-473b-8c4a-f596749143c1)

### Environment

```markdown
- FlutterFlow version: 3.1
- Platform: Chrome Flutterflow desktop app/extension
- Browser name and version: Chrome 114.0.5735.199
- Operating system and version affected: Windows 11 22H2
```

**Comment by Mattmannn** [comment]
https://github.com/FlutterFlow/flutterflow-issues/issues/1007
Found another ticket describing the bug only after posting, from what I understand the bug has been fixed and will be corrected at the next deploy.
